code/modules/mob/living/simple_animal/bot/medbot.dm

```diff
@@ -516,61 +557,80 @@ GLOBAL_VAR(medibot_unique_id_gen)
-						healies *= 1.5
-					if(treatment_method == TOX && HAS_TRAIT(patient, TRAIT_TOXINLOVER))
-						healies *= -1.5
-					if(emagged == 2)
-						patient.reagents.add_reagent(/datum/reagent/toxin/chloralhydrate, 5)
-						patient.apply_damage_type((healies*1),treatment_method)
-						log_combat(src, patient, "pretended to tend wounds on", "internal tools", "([uppertext(treatment_method)]) (EMAGGED)")
-					else
-						patient.apply_damage_type((healies*-1),treatment_method) //don't need to check treatment_method since we know by this point that they were actually damaged.
-						log_combat(src, patient, "tended the wounds of", "internal tools", "([uppertext(treatment_method)])")
-					C.visible_message("<span class='danger'>[src] tends the wounds of [patient]!</span>", \
-						"<span class='userdanger'>[src] tends your wounds!</span>")
+					if(treat_behaviour == "internal_beaker")
+						if(reagent_glass && reagent_glass.reagents.total_volume)
+							var/fraction = min(injection_amount/reagent_glass.reagents.total_volume, 1)
+							var/reagentlist = pretty_string_from_reagent_list(reagent_glass.reagents.reagent_list)
+							log_combat(src, patient, "injected", "beaker source", "[reagentlist]:[injection_amount]")
+							reagent_glass.reagents.reaction(patient, INJECT, fraction)
+							reagent_glass.reagents.trans_to(patient,injection_amount/efficiency, efficiency) //Inject from beaker instead.
+							if(!reagent_glass.reagents.total_volume && !synth_epi)
+								var/list/messagevoice = list("Can someone fill me back up?" = 'sound/voice/medbot/fillmebackup.ogg',"I need new medicine." = 'sound/voice/medbot/needmedicine.ogg',"I need to restock." = 'sound/voice/medbot/needtorestock.ogg')
+								var/message = pick(messagevoice)
+								speak(message,radio_channel)
+								playsound(src, messagevoice[message], 50)
+								declare_cooldown = world.time + 100
+					else if (treat_behaviour == "suck")
+						if(patient.transfer_blood_to(reagent_glass, injection_amount))
+							patient.visible_message("<span class='danger'>[src] is trying to inject [patient]!</span>", \
+								"<span class='userdanger'>[src] is trying to inject you!</span>")
+							log_combat(src, patient, "drained of blood")
+						else
+							to_chat(src, "<span class='warning'>You are unable to draw any blood from [patient]!</span>")
+					C.visible_message("<span class='danger'>[src] injects [patient] with its syringe!</span>", \
+						"<span class='userdanger'>[src] injects you with its syringe!</span>")
 				else
-					tending = FALSE
+					failed = TRUE
 			else
-				tending = FALSE
+				failed = TRUE
 
+			if(failed)
+				visible_message("[src] retracts its syringe.")
 			update_icon()
-			if(!tending)
-				visible_message("[src] places its tools back into itself.")
-				soft_reset()
-		else
-			tending = FALSE
+			soft_reset()
+			return
+
+		treat_behaviour = null
```

Comment:
redundant because you do return at `line 624` (but also, if you remove an indent by the prior request, `line 621` does it.)